satpy/readers/amsr2_l2_gaasp.py

```diff
@@ -0,0 +1,293 @@
+        for is_avail, ds_info in (configured_datasets or []):
+            if is_avail is not None:
+                # some other file handler said it has this dataset
+                # we don't know any more information than the previous
+                # file handler so let's yield early
+                yield is_avail, ds_info
+                continue
+            yield self.file_type_matches(ds_info['file_type']), ds_info
+
+    def _add_lonlat_coords(self, data_arr, ds_info):
+        lat_coord = None
+        lon_coord = None
+        for coord_name in data_arr.coords:
+            if 'longitude' in coord_name.lower():
+                lon_coord = coord_name
+            if 'latitude' in coord_name.lower():
+                lat_coord = coord_name
+        ds_info['coordinates'] = [lon_coord, lat_coord]
+
+    def _get_ds_info_for_data_arr(self, var_name, data_arr):
+        var_suffix = self.filetype_info.get('var_suffix', "")
+        ds_info = {
+            'file_type': self.filetype_info['file_type'],
+            'name': var_name + var_suffix,
+        }
+        x_dim_name = data_arr.dims[1]
+        if x_dim_name in self.dim_resolutions:
+            ds_info['resolution'] = self.dim_resolutions[x_dim_name]
+        if not self.is_gridded and data_arr.coords:
+            self._add_lonlat_coords(data_arr, ds_info)
+        return ds_info
+
+    def _is_2d_yx_data_array(self, data_arr):
+        has_y_dim = data_arr.dims[0] in self.y_dims
+        has_x_dim = data_arr.dims[1] in self.x_dims
+        return has_y_dim and has_x_dim
+
+    def _available_new_datasets(self):
+        possible_vars = list(self.nc.data_vars.items()) + list(self.nc.coords.items())
+        for var_name, data_arr in possible_vars:
+            if data_arr.ndim != 2:
+                # we don't currently handle non-2D variables
+                continue
+            if not self._is_2d_yx_data_array(data_arr):
+                # we need 'traditional' y/x dimensions currently
+                continue
+
+            ds_info = self._get_ds_info_for_data_arr(var_name, data_arr)
+            yield True, ds_info
+
+    def available_datasets(self, configured_datasets=None):
+        """Dynamically discover what variables can be loaded from this file.
+
+        See :meth:`satpy.readers.file_handlers.BaseHandler.available_datasets`
+        for more information.
+
+        """
+        yield from self._available_if_this_file_type(configured_datasets)
+        yield from self._available_new_datasets()
+
+
+class GAASPGriddedFileHandler(GAASPFileHandler):
+    """GAASP file handler for gridded products like SEAICE."""
+
+    y_dims = (
+        'Number_of_Y_Dimension',
+    )
+    x_dims = (
+        'Number_of_X_Dimension',
+    )
+    dim_resolutions = {
+        'Number_of_X_Dimension': 10000,
+    }
+    is_gridded = True
+
+    @staticmethod
+    def _get_extents(data_shape, res):
+        # assume data is centered at projection center
```

Comment:
can we check that?

Reply:
I'm not sure how I would do that? This is essentially missing information from the file producer/creator who, in my opinion, should have just gone with CF conventions for defining the X/Y coordinates. This assumption is correct for the current revision of these files.

Reply:
ok, I was suspecting this. Thanks for clarifying